backend/examples/daemon.rs

```diff
@@ -1,22 +1,22 @@
 use std::env;
-use system76_keyboard_configurator_backend::{run_daemon, Backend};
+use system76_keyboard_configurator_backend::{run_daemon, Backend, Events};
 
 #[cfg(target_os = "linux")]
-fn with_daemon<F: Fn(Backend)>(f: F) {
-    if unsafe { libc::geteuid() == 0 } {
+fn with_daemon<F: Fn(Backend, Events)>(f: F) {
+    let (backend, events) = if unsafe { libc::geteuid() == 0 } {
         eprintln!("Already running as root");
-        let server = Backend::new().expect("Failed to create server");
-        f(server);
-        return;
-    }
+        Backend::new().expect("Failed to create server")
+    } else {
+        Backend::new_pkexec().unwrap()
+    };
 
-    f(Backend::new_pkexec().unwrap());
+    f(backend, events);
 }
 
 #[cfg(not(target_os = "linux"))]
-fn with_daemon<F: Fn(Box<dyn Daemon>)>(f: F) {
-    let server = Backend::new().expect("Failed to create server");
-    f(server);
+fn with_daemon<F: Fn(Backend, Events)>(f: F) {
+    let (backend, events) = Backend::new(false).expect("Failed to create server");
```

Comment:
The bool is no longer necessary when creating the backend. ( #175 )

Reply:
Good catch. Basically all other code here is tested in CI, but this one wasn't since examples/tests are only run on Linux.